rotkehlchen/exchanges/gemini.py

```
@@ -554,9 +594,79 @@ def query_online_deposits_withdrawals(
                 continue
 
             movements.append(movement)
-
+            with self.db.user_write() as write_cursor:
+                if len(history_events) != 0:
+                    db = DBHistoryEvents(self.db)
+                    db.add_history_events(write_cursor=write_cursor, history=history_events)
+                self.db.set_dynamic_cache(
+                    write_cursor=write_cursor,
+                    name=DBCacheDynamic.LAST_QUERY_TS,
+                    value=ts_now(),
+                    location=self.location.serialize(),
+                    location_name=self.name,
+                )
         return movements
 
+    def _deserialize_history_event(
+            self,
+            raw_data: dict[str, Any],
+            timestamp: Timestamp,
+            asset: AssetWithOracles,
+            event_type: AssetMovementCategory,
+        ) -> HistoryEvent | None:
+        """Processes a single transaction from coinbase and deserializes it
+
+        Can log error/warning and return None if something went wrong at deserialization
+        """
+        try:
+            if raw_data.get('status').lower() != 'complete':
+                return None
+            if raw_data.get('type').lower() != 'reward':
+                return None
+
+            amount = deserialize_asset_amount(raw_data.get('amount'))
```

Let the KeyError be raised. You catch it. And it's better than forwarding None to that function.


```suggestion
            amount = deserialize_asset_amount(raw_data[amount'])
```